### PR TITLE
Add a comment explaining the lifecycle of `tls_command_pool_map`.

### PR DESCRIPTION
**Pull request by matanlurey**
Partial work towards https://github.com/flutter/flutter/issues/142083.

Still investigating a fix.

### CODE REVIEW

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 1]
```diff
@@ -158,6 +158,15 @@ void CommandPoolVK::Destroy() {
 // Associates a resource with a thread and context.
 using CommandPoolMap =
     std::unordered_map<uint64_t, std::shared_ptr<CommandPoolVK>>;
+
+// Lifecycle:
+// 1. End of frame will reset the command pool (clearing this on a thread).
+//    There will still be references to the command pool from the uncompleted
+//    command buffers.
+// 2. The last reference to the command pool wuill be destroyed from the fence
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:166]:
```suggestion
// 2. The last reference to the command pool will be released from the fence
```

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 2]
```diff
@@ -158,6 +158,15 @@ void CommandPoolVK::Destroy() {
...
+//    waiter thread, which will schedule something to occur on the resource
```
Comment (jonahwilliams) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:167]:
```suggestion
//    waiter thread, which will schedule a task on the resource
```

File: impeller/renderer/backend/vulkan/command_pool_vk.cc [hunk 3]
```diff
@@ -158,6 +158,15 @@ void CommandPoolVK::Destroy() {
 // Associates a resource with a thread and context.
 using CommandPoolMap =
     std::unordered_map<uint64_t, std::shared_ptr<CommandPoolVK>>;
+
+// Lifecycle:
```
Comment (gaaclarke) [inline impeller/renderer/backend/vulkan/command_pool_vk.cc:162]:
```suggestion
// CommandPoolVK Lifecycle:
```

Anywhere where you talk about a "command pool" I'd switch it to "CommandPoolVK".  We should make it clear that this is referring to the things inside of the `tls_command_pool_map`, not the `tls_command_pool_map` itself.

**Review by gaaclarke — APPROVED** [review]
lgtm!

**Pull request merged**